src/controllers/userController.js

```diff
@@ -123,8 +125,105 @@ class UserController {
       });
     }
   }
+
+  /**
+   * @description - This method is responsible for assigning roles to users
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - object representing response message
+   *
+   * @memberof UserController
+   */
+  static async assignRole(request, response) {
+    try {
+      const { email, role } = request.body;
+      const userToUpdate = await User.findOne({
+        where: {
+          email,
+        }
+      });
+      if (!userToUpdate) {
+        return response.status(404).json({
+          status: 'Fail',
+          message: 'User does not exist'
+        });
+      }
+      if (role !== 'admin' && role !== 'user') {
+        return response.status(400).json({
+          status: 'Fail',
+          message: 'Status can only be "admin" or "user"'
+        });
+      }
+      const fieldToUpdate = {
+        role
+      };
+      const updatedUser = await User.update(fieldToUpdate, {
+        where: {
+          email
+        },
+        returning: true
+      });
+      return response.status(200).json({
+        status: 'Success',
+        message: 'User role updated successfully',
+        updatedUser
+      });
+    } catch (error) {
+      response.status(500).json({
+        status: 'Fail',
+        message: errorResponse(['internal server error, please try again later']),
+      });
+    }
+  }
+
+  /**
+   * @description - This method is responsible for deleting users
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - object representing response message
+   *
+   * @memberof UserController
+   */
+  static async deleteUser(request, response) {
+    const { email } = request.body;
+    try {
+      const userTodelete = await User.findOne({ where: { email } });
```

Comment:
This method `User.findOne()` is repeated in `deleteUser() and assignRole()` controller actions. You can make your codebase modular by extracting the `findOne()` into the middleware or helper file, then call it as a middleware that must be passed before hitting the success controller.
That way, you can always reuse the findOne() as many times as you wish by calling it as a middleware whenever its needed.